node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -183,6 +183,21 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
     if (blockHashes.isEmpty) {
       Future.unit
     } else {
+      for {
+        chainApi <- chainApiFromDb()
+        isIBD <- chainApi.isIBD()
```

Comment:
might be good to have a version of `downloadBlocks` that has the `isIBD` so we don't always need to call `chainApiFromDb` as this does some database reads I believe

Reply:
I don't think the performance hit here is significant. Network latency of transmitting a block will be the limiting factor here.

Reply:
OTOH the fewer disk I/O operations the longer the storage device lives.